jetty-infinispan/infinispan-common/src/main/java/org/eclipse/jetty/session/infinispan/SessionDataMarshaller.java

```diff
@@ -30,13 +39,35 @@
  * control to ensure that session attributes can be deserialized using either
  * the container class loader or the webapp classloader, as appropriate.
  */
-public class SessionDataMarshaller implements MessageMarshaller<InfinispanSessionData>
+public class SessionDataMarshaller implements MessageMarshaller<InfinispanSessionData>, Externalizer<InfinispanSessionData>
 {
     /**
      * The version of the serializer.
      */
     private static final int VERSION = 0;
 
+    private static final Logger LOG = Log.getLogger("org.eclipse.jetty.server.session");
+    
+    private static SerializationContext serializationContext;
+
+    static
+    {
+        FileDescriptorSource fds = new FileDescriptorSource();
+        try
+        {
+            fds.addProtoFiles("/session.proto");
+            serializationContext = ProtobufUtil.newSerializationContext();
+            serializationContext.registerProtoFiles(fds);
+            serializationContext.registerMarshaller(new SessionDataMarshaller());
+        }
+        catch (IOException e)
+        {
+            serializationContext = null;
+            LOG.warn("SerializationContext cannot be initialized", e);
+        }
+
+    }
+    
```

Comment:
The thing is, we don't want to do this static initialization all the time. Certainly when we use infinispan remote, the use of the session.proto is already configured into a different serialization context, so this static initialization is wasted.

In jetty we are generally static-phobic: we prefer if we can do things per instance.